components/dashboard/src/components/PendingChangesDropdown.tsx

```diff
@@ -4,12 +4,56 @@
  * See License.AGPL.txt in the project root for license information.
  */
 
-import { WorkspaceInstance } from "@gitpod/gitpod-protocol";
+import { WorkspaceInstance, WorkspaceInstanceRepoStatus } from "@gitpod/gitpod-protocol";
 import ContextMenu, { ContextMenuEntry } from "./ContextMenu";
 import CaretDown from "../icons/CaretDown.svg";
+import { useQuery } from "@tanstack/react-query";
+
+interface GitStatusResult {
+    status?: WorkspaceInstanceRepoStatus;
+}
+
+const useGitStatusQuery = ({ workspaceInstance }: { workspaceInstance?: WorkspaceInstance }) => {
+    return useQuery<GitStatusResult>({
+        queryKey: [
+            "git-status",
+            workspaceInstance?.id ?? "",
+            workspaceInstance?.status.phase ?? "",
+            workspaceInstance?.ideUrl ?? "",
+        ],
+        queryFn: async () => {
+            if (!workspaceInstance) {
+                return {};
+            }
+            if (workspaceInstance.status.phase !== "running") {
+                return {
+                    status: workspaceInstance.status.repo,
+                };
+            }
+            const workspaceUrl = new URL(workspaceInstance.ideUrl);
+            workspaceUrl.pathname = "/_supervisor/v1/status/git";
+            try {
+                const response = await fetch(workspaceUrl.toString(), { credentials: "include" });
+                if (response.status === 200) {
+                    const result = await response.json();
+                    return result;
```

Comment:
**issue**: Not sure where to comment about this but, could we skip the 0 prefix when changes are less than 10?

<img width="165" alt="Screenshot 2023-07-18 at 12 29 49" src="https://github.com/gitpod-io/gitpod/assets/120486/3e6f4204-9691-47c6-8e98-bdc338c9af87">

Reply:
I think something is up with the JSON encoding of the response in supervisor: fields like `totalUncommitedFiles` should be numbers, not strings. cc @akosyakov 
```json
{
  "status": {
    "branch": "brada",
    "latestCommit": "9c1c4ceebdf760aa952860bbafb3a89e2c6ac669",
    "uncommitedFiles": [],
    "totalUncommitedFiles": "0",
    "untrackedFiles": [],
    "totalUntrackedFiles": "0",
    "unpushedCommits": [],
    "totalUnpushedCommits": "0"
  },
  "lastUpdate": "2023-07-18T09:44:34.438317326Z"
}
```

Reply:
hm, it seems to be expected behaviour of grpc, int64 is bigger than max number in JS :( so it turn to string them and they have to be parsed on client side.

Reply:
or we need to use grpc instead of json

Reply:
I will switch to in32 it is a lot already

Reply:
Let's hope that users stay below 2 billion changed files 😄